### PR TITLE
AI: Multi-strike moves, variable power moves, sleep check

### PR DESCRIPTION
**Pull request by BuffelSaft**
## Description
AI now calculates damage for the following moves/effects:
- Multi-strike moves (except Beat Up and most Parental Bond cases - waiting on #1676 to deal with these, can do that separately)
- Endeavor
- Super Fang/Nature's Madness
- Final Gambit

Parental Bond and type effectiveness are now accounted for when calculating damage for effects like Seismic Toss.

The AI no longer checks type effectiveness when using sleep moves - Dark types aren't immune to Hypnosis and Ghost types aren't immune to Lovely Kiss, so this check isn't needed. I've also renamed `CanBeParayzed` to `AI_CanBeParalyzed`.

## **Discord contact info**
Buffel Saft#2205

### CODE REVIEW

File: src/battle_ai_util.c [hunk 1]
```diff
@@ -799,9 +796,9 @@ s32 AI_CalcDamage(u16 move, u8 battlerAtk, u8 battlerDef)
             dmg = 0;
         // Two hits of Super Fang halves HP twice, leaving target with 25% HP
         else if (AI_DATA->atkAbility == ABILITY_PARENTAL_BOND)
-            dmg = gBattleMons[battlerDef].hp * 3 / 4;
+            dmg = max(1, gBattleMons[battlerDef].hp * 3 / 4);
```
Comment (Xavion3) [inline src/battle_ai_util.c:799]: Should be 2, if we hit twice then we have a minimum of 2 damage as it's 1 damage per hit.

**Comment by Xavion3** [comment]
And yeah, agree that the way IsTwoStrikesMove works currently feels wrong, I'd really like us to not to have a double entry situation where how moves being two hit is determined in different ways in different situations, and the double hit effect doesn't do anything.

**Comment by BuffelSaft** [comment]
Replacing `IsTwoStrikesMove` with a move flag is a good idea, will do that. That way any checks for the double hit effect can just be replaced with a check for the flag and the effect itself can be removed in future.

**Comment by BuffelSaft** [comment]
Fixed a major issue and tidied up the ugly switch case thanks to Syreldar. Seems to be working correctly now:
pokeemerald-0
This is also a good example of why the AI shouldn't give moves an increased score based on effectiveness; here it thinks Seismic Toss and Final Gambit are super effective, so it prefers them over Super Fang, even though Super Fang does almost triple the damage. It's good that it knows Night Shade will have no effect though.

Still need to add the two strikes move flag.

**Comment by BuffelSaft** [comment]
Ready for re-review! Note that the new move flag only acts as a check for the AI; all the extra changes needed for it are done in #1676.

**Review by ghoulslash — APPROVED** [review]

**Pull request merged**